test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -475,7 +483,27 @@ private void wipeCluster() throws Exception {
             waitForPendingRollupTasks();
         }
 
-        final Map<String, List<Map<?,?>>> inProgressSnapshots = wipeSnapshots();
+        // Clean up SLM policies before trying to wipe snapshots so that no new ones get started by SLM after wiping
+        deleteAllSLMPolicies();
+
+        SetOnce<Map<String, List<Map<?,?>>>> inProgressSnapshots = new SetOnce<>();
+        if (waitForAllSnapshotsWiped()) {
+            AtomicReference<Map<String, List<Map<?,?>>>> snapshots = new AtomicReference<>();
+            try {
+                // Repeatedly delete the snapshots until there aren't any
+                assertBusy(() -> {
+                    snapshots.set(wipeSnapshots());
+                    assertThat(snapshots.get().size(), equalTo(0));
```

Comment:
NIT:

```java
assertThat(snapshots.get(), Matchers.anEmptyMap());
```
would print a more useful exception here. We also shouldn't swallow the failure here. It would be better if we would just keep a reference to the `AssertionError` and rethrow it at the end instead of doing another assertion that doesn't contain any information about how we got there?